sklearn/decomposition/_base.py

```diff
@@ -62,7 +62,12 @@ def get_precision(self):
         # handle corner cases first
         if self.n_components_ == 0:
             return np.eye(n_features) / self.noise_variance_
-        if self.n_components_ == n_features:
+
+        # This accounts for both PCA and IPCA
+        observed_samples = (
+            self.n_samples_ if hasattr(self, "n_samples_") else self.n_samples_seen_
+        )
+        if self.n_components_ == min(observed_samples, n_features):
```

Comment:
I didn't know subclasses did not all have `n_samples_`. If that is the case, I'm okay with using `noise_variance_`:

```python
if np.isclose(self.noise_variance_, 0.0, atol=0):
```